addons/l10n_pe_website_sale/static/tests/tours/website_sale_address.js

```diff
@@ -38,3 +38,78 @@ registry.category("web_tour.tours").add('update_the_address_for_peru_company', {
         ...tourUtils.payWithTransfer(),
     ],
 });
+
+registry.category("web_tour.tours").add('maintain_city_district_on_reload', {
+    test: true,
+    url: '/shop',
+    steps: () => [
+        ...tourUtils.addToCart({ productName: "Test Product" }),
+        tourUtils.goToCart({ quantity: 1 }),
+        {
+            content: "Go to checkout",
+            trigger: "a:contains('Checkout')",
+            run: "click",
+        },
+        {
+            content: "Enter an initial value for the State",
+            trigger: 'select[name="state_id"]',
+            run: "text Lima",
+        },
+        {
+            trigger: 'select[name="city_id"]',
+            run: async () => {
+                await new Promise((r) => setTimeout(r, 300));
+            },
+        },
+        {
+            content: "Enter an initial value for the city",
+            trigger: 'select[name="city_id"]',
+            run: "text Lima",
+        },
+        {
+            trigger: 'select[name="l10n_pe_district"]',
+            run: async () => {
+                await new Promise((r) => setTimeout(r, 300));
+            },
+        },
+        {
+            content: "Select an initial value for the district",
+            trigger: 'select[name="l10n_pe_district"]',
+            run: "text Lima",
+        },
+        {
+            content: "Enter an invalid VAT value to trigger a potential reload",
+            trigger: 'input[name="vat"]',
+            run: 'text XAXX010101000',
+        },
+        {
+            content: "Save address",
+            trigger: 'a:contains("Save address")',
+            run: "click",
+        },
+        {
+            content: "Wait for the page to reload",
+            trigger: '.text-danger',
+        },
+        {
+            content: "Check if the city field still has the previously entered value",
+            trigger: 'select[name="city_id"]' ,
+            run: function () {
```

Comment:
Just a detail, but perhaps we could keep the syntax consistent throughout the test, so either use `function` in the steps above (the two async), or the arrow functions here and in the next step, I let you decide